src/compiler/program.ts

```diff
@@ -2187,7 +2187,7 @@ namespace ts {
         }
 
         function createRedirectSourceFile(redirectTarget: SourceFile, unredirected: SourceFile, fileName: string, path: Path, resolvedPath: Path, originalFileName: string): SourceFile {
-            const redirect = createRedirectObject(redirectTarget);
+            const redirect = Object.create(redirectTarget);

```

Comment:
`Object.create` returns `any`, so we should add a type annotation to `redirect` for type safety.